[BUG]KubeBlocks may select new replica itself as DataDump target if none targetPodSelector is set

**Describe the bug**
KubeBlocks may select new replica itself as DataDump target if none targetPodSelector is set, which will cause lifecycle action failed.

**To Reproduce**
Steps to reproduce the behavior:
1. define cmpd with lifecycleAction likes below
```
  lifecycleActions:
    dataDump:
      exec:
        image: {{ .Values.image.registry | default "docker.io" }}/test/test:latest
        command:
        - bash
        - -c
        - |
          set -ex
          echo "start data dump"
        container: mysql
    dataLoad:
      exec:
        image: {{ .Values.image.registry | default "docker.io" }}/test/test:latest
        command:
        - bash
        - -c
        - |
          set -ex
          echo "start data load"
        container: mysql
```
2. horizontal scale the component
3. the new replicas may probably failed data load. `kbagent worker` logs like below
```
> k logs -n thomas-demo t-vip-mysql57-xaza-mysql-vyr1-301 -c kbagent-worker
2025-07-30T17:05:26+08:00	INFO	create service Action	{"actions": "dataDump,dataLoad"}
2025-07-30T17:05:26+08:00	INFO	create service Probe	{"probes": ""}
2025-07-30T17:05:26+08:00	INFO	create service Streaming	{"actions": "dataDump,dataLoad"}
2025-07-30T17:05:26+08:00	INFO	running as worker, tasks: [{"instance":"t-vip-mysql57-xaza-mysql","task":"newReplica","UID":"3","replicas":"t-vip-mysql57-xaza-mysql-vyr1-301","notifyAtFinish":true,"reportPeriodSeconds":60,"newReplica":{"remote":"t-vip-mysql57-xaza-mysql-vyr1-301.t-vip-mysql57-xaza-mysql-headless.thomas-demo.svc.cluster.local","port":3502,"replicas":"t-vip-mysql57-xaza-mysql-vyr1-301"}}]
2025-07-30T17:05:26+08:00	ERROR	failed to run task: {t-vip-mysql57-xaza-mysql newReplica 3 t-vip-mysql57-xaza-mysql-vyr1-301 true 60 0xc0003bf540}, time elapsed: 16.468851ms	{"error": "dial tcp 10.0.0.1:3502: connect: connection refused"}
github.com/apecloud/kubeblocks/pkg/kbagent/service.(*taskService).runTasks
	/src/pkg/kbagent/service/task.go:55
github.com/apecloud/kubeblocks/pkg/kbagent/service.RunTasks
	/src/pkg/kbagent/service/service.go:64
github.com/apecloud/kubeblocks/pkg/kbagent.runAsWorker
	/src/pkg/kbagent/setup.go:246
github.com/apecloud/kubeblocks/pkg/kbagent.Launch
	/src/pkg/kbagent/setup.go:140
main.main
	/src/cmd/kbagent/main.go:87
runtime.main
	/usr/local/go/src/runtime/proc.go:267
panic: failed to run as worker: dial tcp 10.0.0.1:3502: connect: connection refused

goroutine 1 [running]:
main.main()
	/src/cmd/kbagent/main.go:89 +0x3d3
```

**Expected behavior**
`TargetPodSelector` of `Any` and `All` should exclude all not running pods.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Desktop (please complete the following information):**
 - KB branch(release-1.0)

**Additional context**
Add any other context about the problem here.

--- a/controllers/apps/component/transformer_component_workload_ops.go
+++ b/controllers/apps/component/transformer_component_workload_ops.go
@@ -264,6 +264,9 @@ func (r *componentWorkloadOps) buildDataReplicationTask() error {
 
 	// replicas to be created
 	newReplicas := r.desiredCompPodNameSet.Difference(r.runningItsPodNameSet).UnsortedList()
+	if len(newReplicas) == 0 {
+		return nil
+	}
 
 	// replicas in provisioning that the data has not been loaded
 	provisioningReplicas, err := component.GetReplicasStatusFunc(r.protoITS, func(s component.ReplicaStatus) bool {
@@ -273,17 +276,13 @@ func (r *componentWorkloadOps) buildDataReplicationTask() error {
 		return err
 	}
 
-	replicas := append(slices.Clone(newReplicas), provisioningReplicas...)
-	if len(replicas) == 0 {
-		return nil
-	}
-
 	// the source replica
-	source, err := r.sourceReplica(r.synthesizeComp.LifecycleActions.DataDump)
+	source, err := r.sourceReplica(r.synthesizeComp.LifecycleActions.DataDump, provisioningReplicas)
 	if err != nil {
 		return err
 	}
 
+	replicas := append(slices.Clone(newReplicas), provisioningReplicas...)
 	parameters, err := component.NewReplicaTask(r.synthesizeComp.FullCompName, r.synthesizeComp.Generation, source, replicas)
 	if err != nil {
 		return err
@@ -298,16 +297,23 @@ func (r *componentWorkloadOps) buildDataReplicationTask() error {
 	return createOrUpdateEnvConfigMap(transCtx, r.dag, parameters)
 }
 
-func (r *componentWorkloadOps) sourceReplica(dataDump *appsv1.Action) (*corev1.Pod, error) {
+func (r *componentWorkloadOps) sourceReplica(dataDump *appsv1.Action, provisioningReplicas []string) (*corev1.Pod, error) {
 	pods, err := component.ListOwnedPods(r.transCtx.Context, r.cli,
 		r.synthesizeComp.Namespace, r.synthesizeComp.ClusterName, r.synthesizeComp.Name)
 	if err != nil {
 		return nil, err
 	}
+	if len(provisioningReplicas) > 0 {
+		// exclude provisioning replicas
+		pods = slices.DeleteFunc(pods, func(pod *corev1.Pod) bool {
+			return slices.Contains(provisioningReplicas, pod.Name)
+		})
+	}
 	if len(pods) > 0 {
 		if len(dataDump.Exec.TargetPodSelector) == 0 {
 			dataDump.Exec.TargetPodSelector = appsv1.AnyReplica
 		}
+		// TODO: idempotence for provisioning replicas
 		pods, err = lifecycle.SelectTargetPods(pods, nil, dataDump)
 		if err != nil {
 			return nil, err

